### ISSUE TITLE
[3.22.0] Android App cannot build anymore on latest stable flutter version

### ISSUE THREAD

**Issue by AlexDochioiu** [flutter/flutter#148668]
### Steps to reproduce

After running a build/run command such as:
```console
flutter build apk --release --no-tree-shake-icons
```
I started getting the following error after switching to stable flutter 3.22.0 (from 3.19.6): 
```console
Unhandled exception:
RangeError (offset): Invalid value: Not in inclusive range 0..324: 725
#0      RangeError.checkValueInInterval (dart:core/errors.dart:313)
#1      Source.getLocation (package:kernel/ast.dart:14992)
#2      Component.getLocation (package:kernel/ast.dart:14787)
#3      _getLocationInComponent (package:kernel/ast.dart:15181)
#4      Procedure._getLocationInEnclosingFile (package:kernel/ast.dart:3329)
#5      TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#6      TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#7      TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#8      TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#9      TreeNode.location (package:kernel/ast.dart:203)
#10     localFunctionName (package:vm/transformations/type_flow/utils.dart:449)
#11     SummaryCollector.createSummary (package:vm/transformations/type_flow/summary_collector.dart:628)
#12     TypeFlowAnalysis.getSummary (package:vm/transformations/type_flow/analysis.dart:1782)
#13     _DirectInvocation._processFunction (package:vm/transformations/type_flow/analysis.dart:398)
#14     _DirectInvocation.process (package:vm/transformations/type_flow/analysis.dart:287)
#15     _WorkList.processInvocation (package:vm/transformations/type_flow/analysis.dart:1663)
#16     _WorkList.process (package:vm/transformations/type_flow/analysis.dart:1606)
#17     TypeFlowAnalysis.process (package:vm/transformations/type_flow/analysis.dart:1811)
#18     transformComponent (package:vm/transformations/type_flow/transformer.dart:121)
#19     runGlobalTransformations (package:vm/kernel_front_end.dart:589)
#20     compileToKernel (package:vm/kernel_front_end.dart:489)
<asynchronous suspension>
#21     FrontendCompiler.compile (package:frontend_server/frontend_server.dart:639)
<asynchronous suspension>
#22     starter (package:frontend_server/starter.dart:101)
<asynchronous suspension>
#23     main (file:///Volumes/Work/s/w/ir/x/w/sdk/pkg/frontend_server/bin/frontend_server_starter.dart:13)
<asynchronous suspension>

Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileFlutterBuildRelease'.
> Process 'command '/Users/alexandrudochioiu/.asdf/installs/flutter/3.22.0-stable/bin/flutter'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 13s
Running Gradle task 'assembleRelease'...                           14.3s
Gradle task assembleRelease failed with exit code 1
```

### Expected results

App should be able to build/run

### Actual results

App fails to build or run on android

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.22.0, on macOS 14.4.1 23E224 darwin-arm64, locale en-GB)
    • Flutter version 3.22.0 on channel stable at /Users/alexandrudochioiu/.asdf/installs/flutter/3.22.0-stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5dcb86f68f (11 days ago), 2024-05-09 07:39:20 -0500
    • Engine revision f6344b75dc
    • Dart version 3.4.0
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/alexandrudochioiu/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/alexandrudochioiu/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] IntelliJ IDEA Community Edition (version 2022.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.89.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.88.0

[✓] Connected device (3 available)
    • iPhone 15 Pro (mobile)          • 507C41EC-43C2-499A-BCA3-F0B1E40275AA • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-4 (simulator)
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                • darwin         • macOS 14.4.1 23E224 darwin-arm64
    • Chrome (web)                    • chrome                               • web-javascript • Google Chrome 124.0.6367.208

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @AlexDochioiu 
This looks similar to https://github.com/flutter/flutter/issues/138898 and https://github.com/flutter/flutter/issues/140318 which were fixed but need to check if they made it to stable 3.22.0 or not. Meanwhile, can you check if you get same exception on latest master / beta ?

**Comment by AlexDochioiu** [comment]
@darshankawar Thanks for the reply. I think there's not been a beta release newer than this public one? I can try later.

As for `master`. To me it seems to be fully broken at the moment (I already posted a comment here https://github.com/flutter/flutter/issues/148004#issuecomment-2120134716 about it)

**Comment by AlexDochioiu** [comment]
> Thanks for the report @AlexDochioiu This looks similar to #138898 and #140318 which were fixed but need to check if they made it to stable 3.22.0 or not. Meanwhile, can you check if you get same exception on latest master / beta ?

The issue seems indeed similar to those two tickets. Worth saying that unlike them:
a) iOS and web builds with no problems ; only Android build fails
b) On Android, I cannot run the app in either of debug/profile/release. And I also cannot generate an apk/appbundle for it.

**Comment by AlexDochioiu** [comment]
@darshankawar I can confirm now that both latest stable (3.22.0) and latest beta (3.22.0-0.3.pre) fail with the same error when building/running on Android.

I still cannot try to run on `master` because it's either currently broken or something weird is happening on my machine: https://github.com/flutter/flutter/issues/148701

**Comment by darshankawar** [comment]
Thanks for trying @AlexDochioiu. It does seem to be a regression.

I tried the same command in an existing app at my end which built the apk properly so I am unable to replicate it.

```
Running Gradle task 'assembleRelease'...                           90.4s
✓ Built build/app/outputs/flutter-apk/app-release.apk (22.7MB)
dhs@Dhss-MacBook-Pro app_foo_stable % 

```

Is there a way for you to provide us a narrowed down reproducible code sample that triggers the reported error ?

**Comment by AlexDochioiu** [comment]
@darshankawar Not sure I can do that easily. However, I managed to address the issue with my setup causing `master` channel to not work. 

It seems that I can successfully build the android app on master, but not on latest beta or stable.

**Comment by darshankawar** [comment]
Thanks for the update. I'll keep the issue open and label for team's attention on the reported behavior as I am unable to replicate it.

**Comment by mraleph** [comment]
cc @alexmarkov @jensjoha We really need to make this code more defensive and dump more information when we get a failure like this.

**Comment by urim-thummim** [comment]
faced same issue here

**Comment by BenelliFurtado** [comment]
Same here:
* stable 3.22.1 ❌
* beta channel ❌
* master channel 3.23.0-9.0.pre.3 ✅

**Comment by JSBmanD** [comment]
Will it be shipped to the next beta release?

**Comment by workerbee22** [comment]
Same issue here ... looks like possibly related to enforcing Java version?

**Comment by isskj2020** [comment]
I encountered the same error and was able to reproduce it using the following simple code.
The error seems to be caused by the dart-sdk.

```
// file: test_notifier.dart
class TestNotifier extends StateNotifier<String> with LocatorMixin {
  TestNotifier(String state) : super(state);
}

...
// file: main.dart
class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      TestNotifier notifier = TestNotifier("Test");
      _counter++;
    });
  }
```

flutter: 3.22.2 (stable)  / dart: 3.4.3
dependencies: flutter_state_notifier: ^1.0.0

**Comment by jensjoha** [comment]
Fantastic, thanks! I'll see if I can figure out what's happening.

**Comment by jensjoha** [comment]
**TL;DR**
As noted by others above, this crashes on stable but not on main.
After a bisect I believe this to have been fixed by [1e8ea8a7da8be744c41aa2e7d4b12b1bc164aff9](https://github.com/dart-lang/sdk/commit/1e8ea8a7da8be744c41aa2e7d4b12b1bc164aff9).
I will naively assume this will be in the next minor release, flutter 2.23? and dart 3.5? --- maybe @mraleph knows more.

**Details and possible notes for the future**
Reproduced by the above code-snippet from @isskj2020 into a `flutter create` new project. It shortened becomes this:
`main.dart`:
```dart
import 'package:flutter/material.dart';
import 'test_notifier.dart';
void main() => runApp(const MyApp());
class MyApp extends StatelessWidget {
  const MyApp({super.key});
  Widget build(BuildContext context) => const MaterialApp(home: MyHomePage());
}
class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});
  State<MyHomePage> createState() => _MyHomePageState();
}
class _MyHomePageState extends State<MyHomePage> {
  void _incrementCounter() => TestNotifier("Test");
  Widget build(BuildContext context) => Scaffold(
      appBar: AppBar(),
      body: const Center(),
      floatingActionButton: FloatingActionButton(
          onPressed: _incrementCounter, child: const Icon(Icons.add)));
}
```

`test_notifier.dart`:
```dart
import 'package:state_notifier/state_notifier.dart';
class TestNotifier extends StateNotifier<String> with LocatorMixin {
  TestNotifier(super.state);
}
```

`pubspec.yaml` had `state_notifier: ^1.0.0` added as a dependency.

Now `flutter build apk --release --no-tree-shake-icons` crashes when on (current) stable, i.e. 3.22.2.

It can also be run via the dart sdk like this:

```
$DART/out/ReleaseX64/dart $DART/pkg/frontend_server/bin/frontend_server_starter.dart --sdk-root $FLUTTER/bin/cache/artifacts/engine/common/flutter_patched_sdk_product/ --target=flutter --aot --tfa --packages .dart_tool/package_config.json --output-dill app.dill lib/main.dart
```

(assuming the dart checkout and flutter checkout has the same version of files, notably dill version).

Via that [1e8ea8a7da8be744c41aa2e7d4b12b1bc164aff9](https://github.com/dart-lang/sdk/commit/1e8ea8a7da8be744c41aa2e7d4b12b1bc164aff9) doesn't crash, but the commit before does.
The commit also specifically fixes uris related to mixins --- and in the crash `RangeError (offset): Invalid value: Not in inclusive range 0..154: 10068` we notice that
* we're trying to lookup offset 10068 in a file that's only ~154 bytes.
* offset 10068 in `package:state_notifier/state_notifier.dart` points to line `Locator _locator = <T>() => throw DependencyNotFoundException<T>();` in `mixin LocatorMixin` which looks familiar in that `lib/test_notifier.dart`s only class has `with LocatorMixin`.
* `lib/test_notifier.dart` is 153 bytes (I think the added byte is a technicality from the parser).

**Comment by ADez3** [comment]
I have the same issue , I deleted the cache folder on `C:\Users\[usename]\.gradle` and It worked , It seems like there is a problem with android folder in the newly created projects . I hope it gets fixed soon :)

**Comment by BreadKey** [comment]
When archive with Xcode or flutter build ios, same issue appears.

```log```
```
error: Unhandled exception:
RangeError (offset): Invalid value: Not in inclusive range 0..15467: 38979
#0      RangeError.checkValueInInterval (dart:core/errors.dart:313)
#1      Source.getLocation (package:kernel/ast.dart:14992)
#2      Component.getLocation (package:kernel/ast.dart:14787)
#3      _getLocationInComponent (package:kernel/ast.dart:15181)
#4      Procedure._getLocationInEnclosingFile (package:kernel/ast.dart:3329)
#5      TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#6      TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#7      TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#8      TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#9      TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#10     TreeNode._getLocationInEnclosingFile (package:kernel/ast.dart:207)
#11     TreeNode.location (package:kernel/ast.dart:203)
#12     localFunctionName (package:vm/transformations/type_flow/utils.dart:449)
#13     SummaryCollector.createSummary (package:vm/transformations/type_flow/summary_collector.dart:628)
#14     TypeFlowAnalysis.getSummary (package:vm/transformations/type_flow/analysis.dart:1782)
#15     _DirectInvocation._processFunction (package:vm/transformations/type_flow/analysis.dart:398)
#16     _DirectInvocation.process (package:vm/transformations/type_flow/analysis.dart:287)
#17     _WorkList.processInvocation (package:vm/transformations/type_flow/analysis.dart:1663)
#18     _WorkList.process (package:vm/transformations/type_flow/analysis.dart:1606)
#19     TypeFlowAnalysis.process (package:vm/transformations/type_flow/analysis.dart:1811)
#20     transformComponent (package:vm/transformations/type_flow/transformer.dart:121)
#21     runGlobalTransformations (package:vm/kernel_front_end.dart:589)
#22     compileToKernel (package:vm/kernel_front_end.dart:489)
<asynchronous suspension>
#23     FrontendCompiler.compile (package:frontend_server/frontend_server.dart:639)
<asynchronous suspension>
#24     starter (package:frontend_server/starter.dart:101)
<asynchronous suspension>
#25     main (file:///Volumes/Work/s/w/ir/x/w/sdk/pkg/frontend_server/bin/frontend_server_starter.dart:13)
<asynchronous suspension>

Target kernel_snapshot failed: Exception
Failed to package /Users/breadkey/dev/project/pet-space-stationvue.
```
```flutter doctor```
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.2, on macOS 14.5 23F79 darwin-arm64, locale ko-KR)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.90.2)
[✓] Connected device (5 available)
    ! Error: Browsing on the local area network for 정서연 iPhone (2). Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
    ! Error: Browsing on the local area network for iPhone 14 Pro. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
    ! Error: Browsing on the local area network for iPhone 14 Pro. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

• No issues found!
```

**Comment by rafampessoa** [comment]
Same here:

stable 3.22.2 ❌
3.19.6 ✅

**Comment by chplalex** [comment]
The same issue with my project:

beta 3.23.0-0.1.pre ✅
stable 3.22.3 ❌
stable 3.22.2 ❌
stable 3.22.0 ❌
3.19.6 ✅

**Comment by sunflash** [comment]
Both 3.22.2 and 3.22.3 stable don't work for profile and release build, only in debug build. Either go down to 3.19.6 or use 3.23.0-0.1.pre-beta. If you already upgraded all packages and code to use 3.22.2/3, then using 3.23.0-0.1.pre-beta for production build is your only option now. Not ideal but it works ok.

**Comment by ADez3** [comment]
I upgraded the kotlin version on ```
android/settings.gradle
``` from 1.7.10 to 2.0.0 , and it worked for me . Can anyone else try it and give a feedback ?

**Comment by sunflash** [comment]
> I upgraded the kotlin version on `android/settings.gradle` from 1.7.10 to 2.0.0 , and it worked for me . Can anyone else try it and give a feedback ?

I tried it already, kotlin version makes no difference for the issues.

**Comment by ADez3** [comment]
Did you build your app using version 3.22.0 or 3.22.2 ?

**Comment by sunflash** [comment]
@ADez3  We use 3.22.2/3 when debugging, and use 3.23.0-0.1.pre-beta for profile and release build. Have multiple Flutter versions installed on the dev machine.

We have 3 apps in production, and only 1 of 3 can be built successfully with 3.22.2/3 in release mode, and 2 of 3 use 3.23.0-0.1.pre-beta for release build. Can be a bit risky to use "pre-beta" flutter version for the prod release, but it works fine for us, as all our dependencies and code are updated to 3.22.2/3 compatible, and we don't want to revert commits, bugfixes and downgrade dependencies and code down to 3.19.6. Last known good version for production build.

**Comment by sunflash** [comment]
Ideally fix already in "3.23.0-0.1.pre-beta" should be backported down to 3.22.x, so people are not forced to downgrade to 3.19.6 or use 3.23.0-0.1.pre-beta for profile/release build.

**Comment by ADez3** [comment]
Hello guys , Flutter 3.24 is out I tried building an app using it and it works just fine , but with annoying Gradle version warnings, 
can anyone confirm that it works for him also ?

**Comment by chplalex** [comment]
> Hello guys , Flutter 3.24 is out I tried building an app using it and it works just fine , but with annoying Gradle version warnings, can anyone confirm that it works for him also ?

I confirm. My project works with 3.24.0 version well. The issue solved.
